lib/puppet/provider/psrepository/powershellcore.rb

```diff
@@ -56,7 +56,7 @@ def installation_policy=(value)
 
   def flush
     unless @property_flush.empty?
-      flush_command = "Set-PSRepository #{@resource[:name]}"
```

Comment:
Haven't tested it out, but it looks like a breaking change, since you are no longer using the actual name of the resource, making `name` property for every resource mandatory?